src/components/composer/composer/composer.ts

```diff
@@ -397,24 +401,56 @@ export class Composer extends Component<ComposerProps, SpreadsheetChildEnv> {
           text,
           description,
         };
+      })
+      .sort((a, b) => {
+        return a.text.length - b.text.length || a.text.localeCompare(b.text);
       });
+
     if (searchTerm) {
-      values = fuzzyLookup(searchTerm, values, (t) => t.text);
-    } else {
-      // alphabetical order
-      values = values.sort((a, b) => a.text.localeCompare(b.text));
+      values = fuzzyLookup(searchTerm, values, (t) => t.text).slice(0, AUTOCOMPLETE_ENTRIES);
     }
-    this.autoCompleteState.values = values.slice(0, 10);
+
+    this.autoCompleteState.values = values.slice(0, AUTOCOMPLETE_ENTRIES);
+    this.autoCompleteState.getHtmlContent = (value) =>
+      this.getHtmlContentFromPattern(searchTerm, value);
     this.autoCompleteState.selectedIndex = 0;
   }
 
+  private getHtmlContentFromPattern(pattern: string, value: string): HtmlContent[] {
+    const pendingHtmlContent: HtmlContent[] = [];
```

Comment:
This probably should be in the `TextValueProvider` to not pollute the Composer component I'd say

Reply:
I'm not sure, if we do that, we'd have to pass the pattern to `TextValueProvider` so we'd move the whole logic to the child component (to stay consistent) but I don't feel it should be the case, the 'pattern' is no concern of the `TextValueProvider`

Reply:
I'd rather have a `highlighterPattern` props than more things in the composer. Just my opnion tho.

Reply:
that means intrinsically linking the behaviour of `TextValueProvider` with their parent , the highlighterPattern does not make much sense by itself, does it?
That being said, the `getHtmlContentFromPattern` could be a simple helper as it has nothing to do with the composer

Reply:
@LucasLefevre  do you have any thoughts for this?

Reply:
I changed getHtmlContentFromPattern  to be a helper